### PR TITLE
[RFC] :e without a filename is disallowed in terminal buffers

### PR DESCRIPTION
**Pull request by fmoralesc**
Fixes issue #2779, throwing an error as suggested by @tarruda.

### CODE REVIEW

**Comment by justinmk** [comment]
I read https://github.com/neovim/neovim/issues/2779 but I suggest it is silently ignored, rather than showing an error. We support

    :edit term://foo

so it does not make sense to nag the user about `:edit` on an existing `term://`. To fix the immediate problem, we can just silently ignore it. In the future we could consider some other behavior which matches the "refresh" semantics of `:edit` on a normal buffer...?

**Comment by fmoralesc** [comment]
@justinmk This error message should be displayed only when `:edit` receives no arguments. `:edit term://foo` and `:edit file` are unaffected. Maybe I should change the way the message is written, change `filename` with `path spec` or something...

The only valid semantics I see for `:edit in a

**Comment by justinmk** [comment]
> This error message should be displayed only when :edit receives no arguments.

I know. It is inconsistent to call it an error. It should just be ignored.

**Comment by fmoralesc** [comment]
OK. I just removed the error message. I initially thought this should be ignored aswell, I just got confused by what you meant mentioning `edit term://foo`.

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -6570,6 +6570,10 @@ do_exedit (
               )
              || *eap->arg != NUL
              ) {
+    // :e is a no-op in terminal buffers
```
Comment (justinmk) [inline src/nvim/ex_docmd.c]: I think we should write out full commands in comments:

    // :edit ...

I will update this before merging.

**Comment by justinmk** [comment]
> I initially thought this should be ignored aswell, I just got confused by what you meant mentioning edit term://foo

In the future, I think it should re-run `FileType` event if we decide to support that in terminals  (related: https://github.com/neovim/neovim/pull/2419, https://github.com/neovim/neovim/pull/2683), and/or cause a redraw, and possibly other events. Thoughts?

**Comment by justinmk** [comment]
Merged

**Comment by fmoralesc** [comment]
Thanks, @justinmk. 

I hadn't thought of `'filetype'`, but I imagine that could create a bit of an inconsistency in the semantics of the `:edit` command (load `spec` into buffer), and besides, that behaviour can be replicated by the user other ways: `:set filetype=... | redraw`, etc.

**Pull request closed without merging**